pkg/sdn/plugin/node_iptables.go

```diff
@@ -165,23 +152,53 @@ func (n *NodeIPTables) getNodeIPTablesChains() []Chain {
 				{"-i", "docker0", "-m", "comment", "--comment", "from docker to localhost", "-j", "ACCEPT"},
 			},
 		},
-		{
-			table:    "filter",
-			name:     "OPENSHIFT-FIREWALL-FORWARD",
-			srcChain: "FORWARD",
-			srcRule:  []string{"-m", "comment", "--comment", "firewall overrides"},
-			rules: [][]string{
-				{"-s", n.clusterNetworkCIDR, "-m", "comment", "--comment", "attempted resend after connection close", "-m", "conntrack", "--ctstate", "INVALID", "-j", "DROP"},
-				{"-d", n.clusterNetworkCIDR, "-m", "comment", "--comment", "forward traffic from SDN", "-j", "ACCEPT"},
-				{"-s", n.clusterNetworkCIDR, "-m", "comment", "--comment", "forward traffic to SDN", "-j", "ACCEPT"},
-			},
-		},
-		{
+		Chain{
 			table:    "filter",
 			name:     "OPENSHIFT-ADMIN-OUTPUT-RULES",
 			srcChain: "FORWARD",
 			srcRule:  []string{"-i", Tun0, "!", "-o", Tun0, "-m", "comment", "--comment", "administrator overrides"},
 			rules:    nil,
-		},
+		})
+
+	for _, cidr := range n.clusterNetworkCIDR {
+		var masqRule []string
+
+		if n.masqueradeServices {
+			masqRule = []string{"-s", cidr, "-m", "comment", "--comment", "masquerade pod-to-service and pod-to-external traffic", "-j", "MASQUERADE"}
+		} else {
+			masqRule = []string{"-s", cidr, "!", "-d", cidr, "-m", "comment", "--comment", "masquerade pod-to-external traffic", "-j", "MASQUERADE"}
+		}
+
+		chainArray = append(chainArray,
+			Chain{
+				table:    "nat",
+				name:     "OPENSHIFT-MASQUERADE",
```

Comment:
You don't want per-network `Chain` objects, you want a single `"OPENSHIFT-MASQUERADE"` chain, with per-network `rules`